kptncook/mealie.py

```diff
@@ -187,51 +172,265 @@ def login(self, username: str = "admin", password: str = ""):
         access_token = self.fetch_api_token(username, password)
         self.headers = {"authorization": f"Bearer {access_token}"}
 
-    def _post_recipe_and_get_slug(self, recipe):
-        r = self.post("/recipes", data=recipe.json())
+    def get_all_tags(self):
+        all_tags = []
+    
+        r = self.get("/organizers/tags?page=1&perPage=50")
+        r.close()
+        r.raise_for_status()
+        all_tags.extend(r.json()["items"])
+        
+        page = 2
+        while page <= r.json()["total_pages"]:
+            r = self.get(f"/organizers/tags?page={page}&perPage=50")
+            r.close()
+            r.raise_for_status()
+            all_tags.extend(r.json()["items"])
+            page += 1
+
+        #all_tags_parsed = []
+        all_tags_parsed = {}
+        for tag in all_tags:
+            #all_tags_parsed.extend(RecipeTag.parse_obj(tag))
+            all_tags_parsed[tag["name"]] = RecipeTag.parse_obj(tag)
+
+        return all_tags_parsed
+    
+    def get_all_foods(self):
+        all_foods = []
+    
+        r = self.get("/foods?page=1&perPage=50")
+        r.close()
```

Comment:
Uhm, why is close() called here?

Reply:
I had the exception that python can't connect to mealie, only randomly after some time. Since it occurs only after some time, I thought that maybe too many connections were opened at the same time or at least not closed? In the httpx doc I didn't find if I have to close the connection for these request objects myself or if this already happens in one go (I don't think there is connection pooling at this point?) So I just close each connection after using it. After this change I didn't have this problem anymore.

Reply:
The [httpx documentation](https://github.com/encode/httpx/blob/40a0da093b423ca37feee2b85df7c3f0ce3cc96c/httpx/_models.py#L891) states that `response.close()` is called automatically when the body is consumed (`r.json()`).  I've added a context manager to be absolutely sure the connection is closed..